### PR TITLE
cleanup remaining uses of old control specialization interface

### PR DESCRIPTION
**Pull request by anurudhp**
Removes the finally uses of the old `SpecializedSingleQubitControlledExtension` interface in favour of the new interface from #1451.

For controlled versions of `LCUBlockEncoding`, `SelectBlockEncoding` etc, do not update the subbloqs to controlled. Instead, use the appropriate subbloq/controlled-subbloq when decomposing/building call graph.

### CODE REVIEW

**Comment by anurudhp** [comment]
@fdmalone

File: qualtran/bloqs/multiplexers/select_pauli_lcu.py [hunk 1]
```diff
@@ -142,6 +142,9 @@ def get_ctrl_system(self, ctrl_spec: 'CtrlSpec') -> Tuple['Bloq', 'AddControlled
     def adjoint(self) -> 'Bloq':
         return self
 
+    def _has_unitary_(self):
```
Comment (fdmalone) [inline qualtran/bloqs/multiplexers/select_pauli_lcu.py:145]: Why is this necessary in this PR?

File: qualtran/bloqs/mean_estimation/mean_estimation_operator_test.py [hunk 2]
```diff
@@ -63,7 +62,7 @@ class BernoulliEncoder(SelectOracle, SpecializedSingleQubitControlledExtension):
 
     @cached_property
     def control_registers(self) -> Tuple[Register, ...]:
-        return () if self.control_val is None else (Register('control', QBit()),)
+        return ()
```
Comment (fdmalone) [inline qualtran/bloqs/mean_estimation/mean_estimation_operator_test.py:65]: Do we need to keep around this property if it's just empty? (I missed the older PRs so probably need to read those)

**Review by fdmalone — APPROVED** [review]
LGTM

File: qualtran/bloqs/mean_estimation/mean_estimation_operator_test.py (hunk 2 above)
Comment (anurudhp) [reply qualtran/bloqs/mean_estimation/mean_estimation_operator_test.py:65]: `SelectOracle` has this as an abstract property, but as there's no special controlled version I just return `()`.

Ideally the select/prepare interfaces should be updated to the newer `BlockEncoding` style interfaces (would be a major refactor), but as of now as callees use `.control_registers`, I'm keeping it as-is.

File: qualtran/bloqs/multiplexers/select_pauli_lcu.py (hunk 1 above)
Comment (anurudhp) [reply qualtran/bloqs/multiplexers/select_pauli_lcu.py:145]: `cirq.has_unitary` fails to realize that this is unitary because when it decomposes, it gets `And^dagger` gates which aren't unitary (#1488)

**Review by mpharrigan — APPROVED** [review]
nice
